third_party/move/move-vm/runtime/src/storage/module_storage.rs

```diff
@@ -66,20 +66,24 @@ pub trait ModuleStorage: WithRuntimeEnvironment {
 
     /// Returns the metadata in the module, or [None] otherwise. An error is returned if there is
     /// a storage error or the module fails deserialization.
-    fn fetch_module_metadata(
+    ///
+    /// Note: this API is not metered!
+    fn unmetered_get_module_metadata(
         &self,
         address: &AccountAddress,
         module_name: &IdentStr,
     ) -> VMResult<Option<Vec<Metadata>>>;
 
     /// Returns the metadata in the module. An error is returned if there is a storage error,
     /// module fails deserialization, or does not exist.
-    fn fetch_existing_module_metadata(
+    ///
+    /// Note: this API is not metered!
+    fn unmetered_get_existing_module_metadata(
```

Comment:
wondering if it makes sense to only expose `load_module_metadata` and handle whether to charge inside (e.g. having a cache to decide whether it's first access or not)

Reply:
so I added `load_metadata` function which takes meter and decides if charge needs to be done or not, but it is not always available in the caller. In future Aptos VM PRs, we can use `Loader` in AptosVM directly so this should be taken care of